pkg/util/gce/gce.go

```diff
@@ -83,6 +83,19 @@ func GetClusterName() (string, error) {
 	return clusterName, nil
 }
 
+// GetPublicIPv4 returns the public IPv4 address of the current GCE instance
+func GetPublicIPv4() (string, error) {
+	if !config.IsCloudProviderEnabled(CloudProviderName) {
+		return "", fmt.Errorf("cloud provider is disabled by configuration")
+	}
+	clusterName, err := getResponseWithMaxLength(metadataURL+"/instance/network-interfaces/0/access-configs/0/external-ip",
```

Comment:
Can a host have more than one public ip? Same question for the aws implementation.

Reply:
In AWS, the endpoint I'm hitting only returns a single IP. For GCE, we could check for multiple network interfaces, but I stuck with a single IP to match AWS for now. Multiple public IPs is a relatively rare thing, so I don't think it warrants the backend work to support.